dlt/common/data_writers/buffered.py

```diff
@@ -79,10 +81,20 @@ def write_data_item(self, item: TDataItems, columns: TTableSchemaColumns) -> Non
         if isinstance(item, List):
             # items coming in single list will be written together, not matter how many are there
             self._buffered_items.extend(item)
+            # update row count, if item supports "num_rows" it will be used to count items
+            if len(item) > 0 and hasattr(item[0], "num_rows"):
```

Comment:
This is great, makes more sense than counting a million row table as 1 :smile: 

There's an unlikely edge case, you can write a pydantic or dataclass obj with a `num_rows` attribute or method.   
So I would do an `isinstance` check for arrow types or just `if file_format == "arrow" ...`,  since that means we're always dealing with arrow objects, right?